administrative-sdk/connection/connection-controller.js

```diff
@@ -22,48 +22,19 @@ module.exports = class Connection {
     this._analysisId = null;
     this._recordingId = null;
     this._recognitionId = null;
-
-    // The addEventListener interface exists on object.Element DOM elements.
-    // However, this is just a simple class without any relation to the DOM.
-    // Therefore we have to implement a pub/sub mechanism ourselves.
-    // See:
-    // https://developer.mozilla.org/en-US/docs/Web/API/EventTarget.addEventListener
-    // http://stackoverflow.com/questions/10978311/implementing-events-in-my-own-object
-    this.events = {};
+    this.emitter = ee({});
 
     const self = this;
     this.addEventListener = function(name, handler) {
-      if (self.events.hasOwnProperty(name)) {
-        self.events[name].push(handler);
-      } else {
-        self.events[name] = [handler];
-      }
+      self.emitter.on(name, handler);
     };
     this.removeEventListener = function(name, handler) {
-      /* This is a bit tricky, because how would you identify functions?
-       This simple solution should work if you pass THE SAME handler. */
-      if (!self.events.hasOwnProperty(name)) {
-        return;
-      }
-
-      const index = self.events[name].indexOf(handler);
-      if (index !== -1) {
-        self.events[name].splice(index, 1);
-      }
+      self.emitter.off(name, handler);
     };
 
     this.fireEvent = function(name, args) {
-      if (!self.events.hasOwnProperty(name)) {
-        return;
-      }
-      if (!args || !args.length) {
-        args = [];
-      }
-
-      const evs = self.events[name];
-      evs.forEach(ev => {
-        ev(...args);
-      });
+      args = args || [];
```

Comment:
Functions can take default arguments nowadays.

```js
function(name, args=[]) {
   // ...
};
```

I wouldn't mind a fix for this.